src/quotamodel.c

```diff
@@ -911,18 +977,13 @@ merge_uncommitted_table_to_oidlist(List *oidlist)
  */
 
 static void
-calculate_table_disk_usage(bool is_init, HTAB *local_active_table_stat_map)
+calculate_table_disk_usage(bool is_init)
```

Comment:
1) Now the function does not perform incremental quota update. The naming, description contradicts with contents a little bit
2) Why do we put everywhere update_active_table_size and ignore the [case](https://github.com/arenadata/diskquota/blob/767d498fd60664cb27ab2d216521d26ed0a4cb83/src/quotamodel.c#L973-L993) when reloid is not neither in pg_class nor in relation_cache. Or the oidlist from [here](https://github.com/arenadata/diskquota/blob/767d498fd60664cb27ab2d216521d26ed0a4cb83/src/quotamodel.c#L973-L993) does not intersect with oids collected in pull_active_list_from_seg() ? How relevant in this case at all?

Reply:
> Why do we put everywhere update_active_table_size and ignore the [case](https://github.com/arenadata/diskquota/blob/767d498fd60664cb27ab2d216521d26ed0a4cb83/src/quotamodel.c#L973-L993) when reloid is not neither in pg_class nor in relation_cache.

Active tables are always in pg_class, so they don't reach this branch.

Reply:
> the oidlist from [here](https://github.com/arenadata/diskquota/blob/767d498fd60664cb27ab2d216521d26ed0a4cb83/src/quotamodel.c#L973-L993)

What oidlist are you talking about here?

Reply:
> Now the function does not perform incremental quota update. The naming, description contradicts with contents a little bit

There are other functions where the comments don't match the content, see for example pull_active_list_from_seg and pull_active_table_size_from_seg.

Reply:
>There are other functions where the comments don't match the content, see for example

Is it ok?

>so they don't reach this branch.

By the way, what does this branch stands for semantically?

Reply:
> what does this branch stands for semantically?

that branch comes from https://github.com/arenadata/diskquota/pull/27

Reply:
I still can't find the proof that active table oids , which we get from segments, are always in the both pg_class and relation_cache

Reply:
> I still can't find the proof that active table oids , which we get from segments, are always in the both pg_class and relation_cache

This is by definition, the oid of an active table is always in `pg_class` (but not in relation_cache!).

Reply:
>  functions where the comments don't match the content

So, and why don't you want to make things clear at least for functions your patch is changing? And you could write comments for update_active_table_size() as well

Reply:
> why don't you want to make things clear at least for functions your patch is changing? And you could write comments for update_active_table_size() as well

I didn't say that, I'll rename it, correct the comments and add new ones after there are no complaints about the code itself.

Reply:
>I didn't say that, I'll rename it, correct the comments and add new ones after there are no complaints about the code itself.

Ready to approve after this fill be fixed and patch description will contain explanation why its is legal to update quotas right during scanning (https://github.com/arenadata/diskquota/pull/46#discussion_r1897865729, https://github.com/arenadata/diskquota/pull/46#discussion_r1897862799)